web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -258,6 +270,18 @@ private boolean usingAuth() {
         }
     }
 
+    @Cacheable(
+        cacheResolver = "staticRepositoryCacheOneResolver",
+        condition = "@cacheEnabledConfig.getEnabled()"
+    )
+    public List<Sample> fetchSamplesInner(
+        List<String> sampleListIds, String projection
+    ) {
+        return sampleService.fetchSamples(sampleListIds, projection);
```

Comment:
Couple of things.  One, I think we should fetch for samplelistids one by one so that we cache at the list level.  That maskes it more reusable for different combinations of studies.  This will happen a lot.  @jagnathan this would mean fetchSamplesInner would accept on one listId.   @kalletlak points out that caching at the service level here is very redundant because it's just a pass through to repository query, which is itself cached.  This is a good point, however, we want this data to be cached with STATIC cache because the list of possible items is finite.  As we know from experience, when things like this are allowed to be evicted, users really suffer.  Do we have the static option at the repository level of caching?  @Luke-Sikina @kalletlak?

Reply:
I can cache it with a single SampleListId and then add a loop to combine the results for a list of SampleListId.